cpp/include/cudf/detail/reduction.cuh

```diff
@@ -100,7 +100,7 @@ std::unique_ptr<scalar> reduce(InputIterator d_in,
 {
   auto binary_op      = sop.get_binary_op();
   OutputType identity = sop.template get_identity<OutputType>();
-  rmm::device_scalar<OutputType> dev_result{identity, stream, mr};
+  rmm::device_scalar<OutputType> dev_result(identity, stream);
```

Comment:
Why the change from `{}` to `()`? I much prefer the former unless the object has an init-list ctor you're trying to avoid (which device_scalar does not).

Reply:
I vaguely remember @codereport saying that we should prefer `()`, but did not really dig up the convo on Slack to verify. I don't have a preference, may have just misremembered.

Reply:
Unfortunately the guidance is a bit confusing. The rule is to prefer `{}` _unless_ the object you are constructing has an initializer list constructor. E.g.,

```
rmm::device_buffer{0,stream,mr}; // OKAY, device_buffer doesn't have an init list ctor

std::vector<int> v(10); // makes a vector of 10 default initialized ints
std::vector<int> v{10}; // makes a vector with 1 int element initialized to 10
```

This is widely regarded as a pretty significant failing of C++ when they added brace initialization, but it is what it is, and life goes on!

Reply:
Yeah, I'm aware of the issues with {}, so (misremembering Conor's advice) I thought we just want to avoid it, unless init list is actually used.